node-it/src/test/scala/com/wavesplatform/it/sync/AmountAsStringSuite.scala

```diff
@@ -0,0 +1,284 @@
+package com.wavesplatform.it.sync
+
+import com.wavesplatform.account.{AddressOrAlias, KeyPair, PublicKey}
+import com.wavesplatform.common.state.ByteStr
+import com.wavesplatform.it.transactions.BaseTransactionSuite
+import com.wavesplatform.it.api.SyncHttpApi._
+import com.wavesplatform.state.IntegerDataEntry
+import com.wavesplatform.transaction.{CreateAliasTransaction, DataTransaction, TxVersion}
+import com.wavesplatform.transaction.assets.exchange.{AssetPair, ExchangeTransaction, Order}
+import com.wavesplatform.transaction.transfer.MassTransferTransaction.{ParsedTransfer, Transfer}
+import com.wavesplatform.common.utils.{Base58, EitherExt2}
+import com.wavesplatform.it.api.Transaction
+import com.wavesplatform.transaction.Asset.{IssuedAsset, Waves}
+import com.wavesplatform.transaction.assets.SponsorFeeTransaction
+import com.wavesplatform.transaction.transfer.{Attachment, MassTransferTransaction}
+import org.asynchttpclient.Response
+import org.scalatest
+import org.scalatest.Assertion
+import play.api.libs.json.{JsValue, Json}
+
+class AmountAsStringSuite extends BaseTransactionSuite {
+
+  val (headerName, headerValue) = ("Accept", "application/json;large-significand-format=string")
+
+  test("amount as string in assets api") {
+    val assetId = sender.issue(firstAddress, "assetName", "description", someAssetAmount, 8, fee = issueFee, waitForTx = true).id
+    sender.issue(firstAddress, "assetName", "description", quantity = 1, decimals = 0, reissuable = false, fee = issueFee, waitForTx = true).id
+    val currentHeight   = sender.height
+    sender.assetsDetails(assetId, amountsAsStrings = true).quantity shouldBe someAssetAmount
+    sender.assetBalance(firstAddress, assetId, amountsAsStrings = true).balance shouldBe someAssetAmount
+    sender.assetsBalance(firstAddress, amountsAsStrings = true).balances.head.balance shouldBe someAssetAmount
+    sender.nftAssetsBalance(firstAddress,1, amountsAsStrings = true).head.quantity shouldBe 1
+
+    sender.waitForHeight(currentHeight + 1)
```

Comment:
why?

Reply:
because if you request assetDistribution on current height you get "{"error":199,"message":"Using 'assetDistributionAtHeight' on current height can lead to inconsistent result"}" error